File: frontend/src/types/entities/warehouse.ts
/**
 * Canonical operational Warehouse entity returned by the backend POS
 * warehouse-read contract.
 */
export interface Warehouse {
  readonly id: string;

  readonly branch_id: string;

  readonly code: string;

  readonly name: string;

  readonly warehouse_type: string;

  readonly is_active: boolean;
}

File: app/serializers/warehouse.py
def serialize_warehouse(warehouse) -> dict:
    return {
        "id": str(warehouse.id),
        "branch_id": str(warehouse.branch_id),
        "code": warehouse.code,
        "name": warehouse.name,
        "warehouse_type": warehouse.warehouse_type,
        "is_active": bool(warehouse.is_active),
    }
